### ISSUE TITLE
Blessing a vertex breaks graphs

### ISSUE THREAD

**Issue by merkys** [graphviz-perl/Graph#27]
This is probably a very rare case, but I bumped into it recently:
```Perl
use strict;
use warnings;
use Graph::Undirected;

{
    package Wrapper;
    sub new { return bless $_[1] }
}

$\ = "\n";

my $g = Graph::Undirected->new;

my $A = {};
my $B = {};

$g->add_edge( $A, $B );
print $g;

my $C = Wrapper->new( $A );
print $g; # This shows the same vertex appearing twice in the graph
print 'vertices: ', scalar $g->vertices; # OK
print 'edges:    ', scalar $g->edges;    # OK
print 'degrees:';
for ($g->vertices) {
    print scalar $g->neighbours( $_ ); # Should be 1 and 1, but prints 0 and 1
}
```
It seems that blessing a vertex breaks the adjacency table. The blessed reference does not change its address in the memory (`0x...` stays the same), but it changes its type. I assume that in Graph object vertices are identified by their string representation, `HASH(0x...)`, which after blessing becomes `Wrapper=HASH(0x...)` resulting in broken adjacency tables.

It might be that this will not fix in the general case. If so, it would be nice to have a caveat message for this.

**Edit**: Observed on Graph 0.9724.

**Comment by mohawk2** [comment]
This is absolutely a situation where you need to use `refvertexed` (see the docs). Annoyingly, that didn't correctly work here due to a bug (as you can see in the commit linked here).

Thanks for the excellent report which made this easy to reproduce! Now released as 0.9725.

**Comment by merkys** [comment]
Many thanks for a prompt fix! Indeed, with `refvertexed` I now get the correct results.
